certora/helpers/Utils.sol

```diff
@@ -32,4 +33,16 @@ contract Utils {
     function factory(address adapter) external view returns (address) {
         return IReturnFactory(adapter).factory();
     }
+
+    function decodeMarketParams(bytes memory data) external pure returns (MarketParams memory) {
+        return abi.decode(data, (MarketParams));
+    }
+
+    function id(MarketParams memory marketParams) external pure returns (Id) {
+        return MarketParamsLib.id(marketParams);
+    }
+
+    function havocAll() external {
+        this.havocAll();
```

Comment:
is this to detect actual use of the function?

Reply:
This is a helper function to summarize as `HAVOC_ALL`. The fact that it calls itself (indefinitely) is useful because in CVL direct calls to a function are not summarized. So, if in a rule you call this function, and you summarized it with `HAVOC_ALL`, it will:
- call the function directly (not summarize it)
- this function calls itself, which is considered and indirect call, and this time it will be summarized